src/SourceBuild/content/build.proj

```diff
@@ -43,6 +53,132 @@
       VmrBuildNumber="$(BUILD_BUILDNUMBER)" />
   </Target>
 
+  <PropertyGroup>
+    <ExtractDotNetSdkArchiveDependsOn>SetupScenarioTestsArtifacts;GetDotNetSdkArtifactPath</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.zip'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkZip</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.tar.gz'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkTarball</ExtractDotNetSdkArchiveDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ExtractDotNetSdkArchive"
+          DependsOnTargets="$(ExtractDotNetSdkArchiveDependsOn)" />
+
+  <PropertyGroup>
+    <_ScenarioTestsNuGetConfig>$(ScenarioTestsArtifactsDir)NuGet.config</_ScenarioTestsNuGetConfig>
+    <_InstallerNuGetConfig>$(RepoRoot)src/installer/NuGet.config</_InstallerNuGetConfig>
+  </PropertyGroup>
+
+  <Target Name="SetupNuGetConfig"
+          Inputs="$(_InstallerNuGetConfig)"
+          Outputs="$(_ScenarioTestsNuGetConfig)">
+
+    <Copy SourceFiles="$(_InstallerNuGetConfig)"
+          DestinationFiles="$(_ScenarioTestsNuGetConfig)" />
+
+    <AddSourceToNuGetConfig NuGetConfigFile="$(_ScenarioTestsNuGetConfig)"
+                            SourceName="local-feed"
+                            SourcePath="$(ScenarioTestsLocalFeedDir)" />
+  </Target>
+
+  <ItemGroup>
+    <_ArtifactsPackage Include="$(ArtifactsShippingPackagesDir)**;
+                                $(ArtifactsNonShippingPackagesDir)**"
+                       Condition="!$([System.String]::Copy('%(Identity)').EndsWith('.symbols.nupkg'))" />
+  </ItemGroup>
+
+  <Target Name="CreateScenarioTestsLocalFeed">
+
+    <RemoveDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <MakeDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <!-- Combine all packages into a single directory with symlinks so that we can have a single feed that points to them all. -->
+    <Copy SourceFiles="@(_ArtifactsPackage)"
+          DestinationFolder="$(ScenarioTestsLocalFeedDir)"
+          UseSymbolicLinksIfPossible="true" />
+
+  </Target>
```

Comment:
Why can't scenario-tests use the same infrastructure to flow dependencies across inner-repositories?

Reply:
Assuming we don't need non-shipping packages for testing, I don't think the repo dependency flow structure applies here. It would also be awkward, since you'd have to introduce all the dependencies into the V.D.xml file, when the repo build wouldn't actually depend on them to build (there would be no use of the version properties).

Reply:
Scenario-tests should be able to consume the live built non-shipping packages, i.e. Microsoft.DotNet.XUnitExtensions which we want to use there.

Reply:
> How will scenario-tests know which version of runtime packages to reference when building the tests?

The test infra has a nuget feed config like any other, and the tests themselves (typically templatles) are doing sdk operations. The SDK is the one defining what versions should be used.